docs/build-for-developers/cli-usage.md

```diff
@@ -54,10 +55,39 @@ openfn path/to/job.js -a http=/repo/openfn/adaptors/my-http-build
 openfn path/to/job.js -ma http
 ```
 
-Set a path to the monorepo with the env var OPENFN_REPO_DIR (eg, `OPENFN_REPO_DIR=~/openfn/repo openfn job.js -ma http`).
+Set a path to the monorepo with the env var OPENFN_REPO_DIR (eg,
+`OPENFN_REPO_DIR=~/openfn/repo openfn job.js -ma http`).
 
 Remember to rebuild the adaptor before using it!
 
+**Run from a specific start step**
+
+You can specify a step as an exact id, or a partial substring from the name or
+id.
+
+```bash
+openfn path/to.job.js --start cf628d9e -s path/to/input.json
+```
+
+If you have previously cached this workflow's results, the CLI will
+automatically load the correct input from the cache if you omit the `-s`
+argument:
+
+```bash
+openfn path/to.job.js --start cf628d9e
+```
+
+You can also pass `--end` to make the workflow exit early.
+
+**Run a single step**
+
+`--only` works just like `--start` and `--end`. You can partially match a step
+name or id, and input will be automatically loaded from the cache.
+
+```bash
+openfn path/to.job.js --only cf628d9e
```

Comment:
I am just curious if i don't have cli cache, will this brow up `openfn path/to.job.js --only cf628d9e` i am curios if we need to emphasis on this will only work if you previous use `--cache-steps`

Reply:
`--cache-steps` and `--only` are mutually exclusive and technically unrelated.

If you do `--only` without having cached steps:
- You can pass your own input with `-s`, which is likely what I'd expect you to do
- If you didn't pass state and the CLI couldn't find anything in the cache, you'll get a warning
- If the step doesn't need input anyway, it'll run just fine